### ISSUE TITLE
Return a sensible status message on /api/health while starting up

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#4570]
Especially when waiting on migrations, we should let the user know something is happening. Right now there's no real way to distinguish between a dead instance and one that is migrating itself.

**Comment by camsaul** [comment]
we're already doing that 😉  Here's the definition of the health endpoint

```clojure
(if ((resolve 'metabase.core/initialized?))
  {:status 200, :body {:status "ok"}}
  {:status 503, :body {:status "initializing", :progress ((resolve 'metabase.core/initialization-progress))}})
```

**Comment by camsaul** [comment]
So noted in a Slack discussion there's a couple issues we need to fix here:

*  We try to check `(metabase.core/initialized?)` before the namespace is necessarily loaded, which makes the endpoint barf. We should fix that by moving the initialization var into a namespace that can get loaded ASAP and then add it as a `:require`-ment instead of trying to hackily `resolve` the var
*  it looks like the initialization status progress doesn't take into account whether the DB is finished initializing (?) or at least it's possible to hit the other API endpoints before the DB is set up. For example, visiting the home page will fail and giving you a generic 500 `DB is not set up` error message. Should we return a better error message instead (or redirect you to some other page?) while we're waiting for DB setup to finish?

**Comment by salsakran** [comment]
We should definitely do the first at least. 

The second ideally, should have something less dire than a 500 with a cryptic clojure error. A better error message would be sufficient.

**Comment by salsakran** [comment]
At least one person thought this was a failed upgrade -

http://discourse.metabase.com/t/how-do-i-update-metabase-running-on-elastic-beanstalk/53/13

**Comment by camsaul** [comment]
Fixed by #4614
